lib/active_model/serializer/associations.rb

```diff
@@ -74,7 +74,8 @@ def has_one(name, options = {}, &block) # rubocop:disable Style/PredicateName
         # @api private
         #
         def associate(reflection)
-          self._reflections << reflection
+          key = reflection.options[:key]
+          key ? self._reflections[key] = reflection : self._reflections[reflection.name] = reflection
```

Comment:
because you have assignments in the true/false areas of ternary.

proper form has has the ternary only produce the value, and not do any mutations.

So, the only way to re-write the ternary without breaking that rule would be to convert it to an if block, unless you did:

```ruby
key_name = key ? key : reflection.name
self._reflections[key_name] = reflection
```